src/libutil/include/nix/util/util.hh

```diff
@@ -58,6 +58,14 @@ Strings quoteStrings(const C & c, char quote = '\'')
     return res;
 }
 
+inline Strings quoteFSPaths(std::set<std::filesystem::path> paths, char quote = '\'')
+{
+    Strings res;
+    for (auto & s : paths)
+        res.push_back(quoteString(s.string()));
+    return res;
+}
```

Comment:
`std::ranges` magic for this, and if it is short enough we can just inline it

Reply:
done, lmk what you think. doesn't look more succinct to me

Reply:
I like the new way, even though it is kinda ridiculous :)